taskvine/src/manager/vine_schedule.c

```diff
@@ -211,9 +204,42 @@ int check_worker_against_task(struct vine_manager *q, struct vine_worker_info *w
 		}
 	}
 
+	if (t->needs_library) {
+		struct vine_task *library = vine_schedule_find_library(w, t->needs_library);
+		if (library) {
+			if (library->function_slots_inuse >= library->function_slots) {
+				return 0;
+			}
+		} else {
+			/* this is bad and we should feel bad. checking for matches should not modify the state of
+			 * workers. */
+			library = send_library_to_worker(q, w, t->needs_library);
+			/* Careful: If this failed, then the worker object may longer be valid! */
+			if (!library) {
+				return 0;
+			}
+		}
+	}
+
 	return 1;
 }
 
+/* Find a library task running on a specific worker that has an available slot.
+ * @return pointer to the library task if there's one, 0 otherwise. */
+struct vine_task *vine_schedule_find_library(struct vine_worker_info *w, const char *library_name)
```

Comment:
Does this need to check for available slots on the library?
Otherwise it could return a library that is already fully occupied.

Reply:
That's fine. We check for slots somewhere else. And that was the bug, a library occupied was 'not found', so the library was sent again.